scripts/release.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+set -e
+
+hash git &> /dev/null
+if [[ $? -eq 1 ]]; then
+    echo >&2 "git is not installed"
+fi
+
+if [[ -z $1 || -z $2 ]]; then
+  echo "This script will create a release branch and merge it into main"
+  echo ""
+  echo "Syntax: release.sh {command} {version}"
+  echo ""
+  echo "commands:"
+  echo -e "\trelease\t\tcreate a release branch"
+  echo -e "\tmain\t\tmerge a release branch into main"
+  echo ""
+  echo "version:"
+  echo -e "\tmust be in the format vX.Y.Z, e.g. v1.0.0"
+  exit
+fi
+
+if [[ -n $(git status -s) ]]; then
+  echo "Cannot continue because your repository is dirty"
+  exit
+fi
+
+echo "Fetching latest repository state..."
+git fetch &> /dev/null
```

Comment:
do we want to hide the error output? what happens if auth fails?

Reply:
the script crashes